functions/gbs_import_functions.php

```diff
@@ -279,6 +279,83 @@ function gbs_import_sequenza($open_data_file_handle, $sample) {
 	return $genome_block_store;
 }
 
+#############################################
+# GBS IMPORT FROM PURPLE FILE
+#############################################
+
+function gbs_import_purple($open_data_file_handle, $sample) {
+	// Create an array to store block information
+	$genome_block_store = array();
+	
+	// Go through the file line by line
+	while (($line = fgets($open_data_file_handle)) !== false) {
+		// Split the row into an array by column
+		$columns = explode("\t", $line);
+		
+		// If it's an empty line, ignore it
+		if (count($columns) == 0) {
+			continue;
+		}
+		
+		// Ignore the header line
+		if ($columns[0] == "#chromosome") {
+			continue;
+		}
+		
+		// If the line does not have the expected number of columns, quit out and display it to the user
+		if (count($columns) != 10) {
+			// Save the line to display to the user
+			log_gbs_import_info("error", "A row in your PURPLE output file did not contain the right amount of columns, it should be 10. Line: ".$line);
+			
+			return false;
+		}
+		
+		// Remove space/newline characters from the 10th column (the last one which has \n at the end)
+		$columns[9] = preg_replace("/[\n\r\s]*/", "", $columns[9]);
+		
+		// Make sure the coordinate and copy number columns are numeric
+		if (!is_numeric($columns[1]) || !is_numeric($columns[2]) || !is_numeric($columns[3])) {
+			// Save the line to display to the user
+			log_gbs_import_info("error", "Found a line which contains one or more non-numeric values for coordinates or copy number. Line: ".$line);
+			
+			return false;
+		}
+		
+		// Ignore all events that are too close to a copy number of 2
+		if ($columns[3] < 2.5 && $columns[3] > 1.5) {
```

Comment:
Please disable filtering upon import. let's add ploidy slider, defaulting to the same values for filtering at runtime. all small variants will now be tagged with the ploidy of the overlapping segment(s).